packages/api/src/router/user.ts

```diff
@@ -1,3 +1,32 @@
-import { createTRPCRouter } from "../trpc";
+import { clerkClient } from "@clerk/nextjs";
+import { z } from "zod";
 
-export const userRouter = createTRPCRouter({});
+import { adminProcedure, createTRPCRouter, protectedProcedure } from "../trpc";
+
+export const userRouter = createTRPCRouter({
+  getAllUsers: protectedProcedure.query(async ({ ctx }) => {
+    return await ctx.db.user.findMany();
+  }),
+
+  // Fetches the clerk data of every user that has signed up and returns a mapping of the form:
+  // <clerkId, userData>
+  getAllClerkUsers: protectedProcedure.query(async () => {
+    const users = await clerkClient.users.getUserList();
+    return new Map(users.map((user) => [user.id, user]));
+  }),
```

Comment:
should non-admin users have access to this information?

Reply:
I think it's fine, since under some circumstances we may need all the information even if we're not in the admin dashboard (if we have a drop down that needs show all the users for example)

Reply:
I'm not sure if we have any usecases like that right now 🤔  -- would recommend that we keep it admin right now, then just move to protected if we need to in the future